maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/SpaInstaller.java

```diff
@@ -0,0 +1,122 @@
+package org.openmrs.maven.plugins.utility;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.BuildPluginManager;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.project.MavenProject;
+import org.json.simple.JSONObject;
+import org.openmrs.maven.plugins.model.DistroProperties;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class SpaInstaller {
+
+    static final String BAD_SPA_PROPERTIES_MESSAGE = "Distro properties file contains invalid 'spa.' elements. " +
+            "Please check the distro properties file and the specification. " +
+            "Parent properties cannot have their own values (i.e., if 'spa.foo.bar' exists, 'spa.foo' cannot be assigned a value). " +
+            "Duplicate properties are not allowed.";
+    static final String NODE_VERSION = "16.2.0";
+    static final String NPM_VERSION = "7.6.2";
```

Comment:
Node 16.2.0 was bundled with NPM 7.13.0. Is there a reason to keep pinned to the older version?